sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/AzureTableImplTest.java

```diff
@@ -0,0 +1,521 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.tables;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.AddDatePolicy;
+import com.azure.core.http.policy.AddHeadersPolicy;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.test.TestBase;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.tables.implementation.AzureTableImpl;
+import com.azure.data.tables.implementation.AzureTableImplBuilder;
+import com.azure.data.tables.implementation.models.OdataMetadataFormat;
+import com.azure.data.tables.implementation.models.QueryOptions;
+import com.azure.data.tables.implementation.models.ResponseFormat;
+import com.azure.data.tables.implementation.models.TableProperties;
+import com.azure.storage.common.implementation.connectionstring.StorageAuthenticationSettings;
+import com.azure.storage.common.implementation.connectionstring.StorageConnectionString;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+
+/**
+ * This class tests the Autorest code for the Tables track 2 SDK
+ */
+public class AzureTableImplTest extends TestBase {
+    private static final String PARTITION_KEY = "PartitionKey";
+    private static final String ROW_KEY = "RowKey";
+    private static final int TIMEOUT = 5000;
+    private AzureTableImpl azureTable;
+
+    @Override
+    protected void beforeTest() {
+        super.beforeTest();
+        String connectionString = interceptorManager.isPlaybackMode()
+            ? "DefaultEndpointsProtocol=https;AccountName=dummyAccount;AccountKey=xyzDummy;EndpointSuffix=core.windows.net"
+            : System.getenv("AZURE_TABLES_CONNECTION_STRING");
+        StorageConnectionString storageConnectionString
+            = StorageConnectionString.create(connectionString, new ClientLogger(AzureTableImplTest.class));
+
+        StorageAuthenticationSettings authSettings = storageConnectionString.getStorageAuthSettings();
+        TablesSharedKeyCredential sharedKeyCredential = new TablesSharedKeyCredential(authSettings.getAccount().getName(),
+            authSettings.getAccount().getAccessKey());
+
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
```

Comment:
I'd be consistent about using `final`. If you use final here. I'd expect you to use it in all other cases in your code where you don't expect variables to be reassigned.